### ISSUE TITLE
Swagger 1.5 fails to start with AIOOBE

### ISSUE THREAD

**Issue by Plonk42** [swagger-api/swagger-core#1282]
Hi there,

I have this exception when starting my application with Swagger.

```java
java.lang.ArrayIndexOutOfBoundsException: 0
	at io.swagger.jaxrs.utils.ReflectionUtils.findMethod(ReflectionUtils.java:93)
	at io.swagger.jaxrs.utils.ReflectionUtils.getOverriddenMethod(ReflectionUtils.java:58)
	at io.swagger.jaxrs.Reader.getAnnotation(Reader.java:865)
	at io.swagger.jaxrs.Reader.parseMethod(Reader.java:701)
	at io.swagger.jaxrs.Reader.read(Reader.java:286)
	at io.swagger.jaxrs.Reader.read(Reader.java:169)
	at io.swagger.jaxrs.Reader.read(Reader.java:146)
	at io.swagger.jaxrs.config.BeanConfig.setScan(BeanConfig.java:170)
[...]
```

It is not always, but I would say 80% of the time, which make me think of a race-condition somewhere.

I'm using

```xml
		<dependency>
			<groupId>io.swagger</groupId>
			<artifactId>swagger-jersey2-jaxrs</artifactId>
			<version>1.5.0</version>
		</dependency>
```

Thanks!

**Comment by Plonk42** [comment]
Here is `findMethod()` in `io.swagger.jaxrs.utils.ReflectionUtils`:

```java
    public static Method findMethod(Method methodToFind, Class<?> cls) {
        String methodToSearch = methodToFind.getName();
        Class<?>[] pTypes = methodToFind.getParameterTypes();
        Type[] gpTypes = methodToFind.getGenericParameterTypes();
        methodLoop:
        for (Method method : cls.getMethods()) {
            if (!method.getName().equals(methodToSearch) || !method.getReturnType().isAssignableFrom(methodToFind.getReturnType())) {
                continue;
            }
            Class<?>[] pt = method.getParameterTypes();
            Type[] gpt = method.getGenericParameterTypes();
            for (int j = 0; j < pTypes.length; j++) {
                Class<?> parameterType = pTypes[j];
                if (!(pt[j].equals(parameterType) || (!gpt[j].equals(gpTypes[j]) && pt[j].isAssignableFrom(parameterType)))) {
                    continue methodLoop;
                }
            }
            return method;
        }
        return null;
    }
```

Line 93 being:
```java
                if (!(pt[j].equals(parameterType) || (!gpt[j].equals(gpTypes[j]) && pt[j].isAssignableFrom(parameterType)))) {
```

How can we be sure that `pt.size() == pTypes.size()` ?
I think we should first check that the candidate method has the same number of parameters than the one we are looking for. Maybe in the same place where we check name and return type (line 86):

```java
            if (!method.getName().equals(methodToSearch) || !method.getReturnType().isAssignableFrom(methodToFind.getReturnType())) {
                continue;
            }
```

What do you think?

**Comment by Plonk42** [comment]
I confirm the issue comes from two overrided methods with same name and different parameters.

I solved the issue by renaming a method, moving from:
```java
	@Override
	@GET
	public Response get() {
		return _get();
	}

	@Override
	@GET
	@Path("{identifier}")
	public Response get(@PathParam("identifier") ResourceIdentifierDto identifier)  {
		return _get(identifier);
	}
```

to:

```java
	@Override
	@GET
	public Response getAll() {
		return _get();
	}

	@Override
	@GET
	@Path("{identifier}")
	public Response get(@PathParam("identifier") ResourceIdentifierDto identifier)  {
		return _get(identifier);
	}
```
About the fact that it is not always reproducible, it is because  `java.lang.Class.getMethods()` ordering is not guaranteed, cf. javadoc:

> The elements in the array returned are not sorted and are not in any particular order.

**Comment by webron** [comment]
@Plonk42 - thanks for submitting the issue and investigating it. Can you try with the 1.5.1-SNAPSHOT? That specific piece of code has changed a bit.

**Comment by webron** [comment]
@Plonk42 - did you get a chance to test it?

**Comment by Plonk42** [comment]
Not yet, I've renamed a method in my code as a workaround so I moved to something else. 
I had a quick look at 1.5.1-SNAPSHOT and indeed the code id modified in such a way I think this bug is gone. But again, and did not test it...

**Comment by webron** [comment]
Okay, we've released 1.5.1 officially yesterday, so you can test it directly (without a snapshot).

**Comment by webron** [comment]
Closing due to inactivity. Please reopen if needed.
